### PR TITLE
Set AP-Password

### PR DESCRIPTION
**Pull request by RuffaloLavoisier**

### CODE REVIEW

**Review by RuffaloLavoisier — COMMENTED** [review]
Looks good to me.

File: include/locales/de-DE.h [hunk 1]
```diff
@@ -21,6 +21,8 @@
 #define LANG_WEBSRV_TITLE "Einstellungen"
 #define LANG_WEBSRV_USER "Nutzer:"
 #define LANG_WEBSRV_PASS "Passwort:"
+#define LANG_WEBSRV_AP_PASSWORD_ON "AP pw ON"
```
Comment (uvwxy) [inline include/locales/de-DE.h:24]: ON = an
OFF = aus

File: src/apps/tools/OswAppWebserver.cpp [hunk 2]
```diff
@@ -18,14 +18,21 @@ void OswAppWebserver::loop() {
...
     if (OswServiceAllTasks::wifi.isConnected()) {
         hal->gfx()->print(LANG_DISCONNECT);
     } else {
         if(OswServiceAllTasks::wifi.isEnabled())
             hal->gfx()->print("...");
-        else
+        else {
             hal->gfx()->print(LANG_CONNECT);
+            OswUI::getInstance()->setTextCursor(BUTTON_2);
+            if(OswConfigAllKeys::hostPasswordEnabled.get())
```
Comment (uvwxy) [inline src/apps/tools/OswAppWebserver.cpp:31]: Please, either use braces `if(..){...}else{...}` or ternary expression.

File: src/services/OswServiceTaskWiFi.cpp [hunk 3]
```diff
@@ -74,7 +74,10 @@ void OswServiceTaskWiFi::loop() {
             if(this->m_connectFailureCount % 4 == 3) {
                 if(OswConfigAllKeys::wifiAutoAP.get()) {
                     if(!this->m_enableStation) {
-                        this->enableStation();
+                        if (OswConfigAllKeys::hostPasswordEnabled.get())
```
Comment (uvwxy) [inline src/services/OswServiceTaskWiFi.cpp:77]: `if(..){...}else{...}`

File: src/services/OswServiceTaskWiFi.cpp [hunk 4]
```diff
@@ -292,6 +295,15 @@ void OswServiceTaskWiFi::disableStation() {
 #endif
 }
 
+void OswServiceTaskWiFi::switchingAPpassword(){
+    OswConfig::getInstance()->enableWrite();
+    OswConfigAllKeys::hostPasswordEnabled.set(!OswConfigAllKeys::hostPasswordEnabled.get());
+#ifndef NDEBUG
```
Comment (uvwxy) [inline src/services/OswServiceTaskWiFi.cpp:301]: NDEBUG or DEBUG? (maybe I missed something?)

Comment (simonmicro) [reply src/services/OswServiceTaskWiFi.cpp:301]: Yes.

https://github.com/Open-Smartwatch/open-smartwatch-os/blob/a2d2967fd08e469dbe7d4fc56c1d88c7b92d04b8/scripts/build/prebuild_cppflags.py#L3

**Pull request merged**